src/MongoDB.Driver/Encryption/MongocryptdFactory.cs

```diff
@@ -167,6 +167,8 @@ private void StartProcess(string path, string args)
                     process.StartInfo.FileName = path;
                     process.StartInfo.CreateNoWindow = true;
                     process.StartInfo.UseShellExecute = false;
+                    process.StartInfo.RedirectStandardOutput = true;
+                    process.StartInfo.RedirectStandardError = true;
```

Comment:
OK. Thanks for the explanation.

Reply:
Do we need to worry about where standard out and standard error are pointing? Do we need to explicitly redirect these streams to whatever the Windows equivalent of `/dev/null` is?

Reply:
Interesting!
Fwiw, a brief reading of https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.redirectstandardoutput?redirectedfrom=MSDN&view=netframework-4.8#System_Diagnostics_ProcessStartInfo_RedirectStandardOutput seems to indicate that the output of mongocryptd is stored in `process.StandardOutput` and `process.StandardError`

Reply:
>that the output of mongocryptd is stored in process.StandardOutput and process.StandardError

yes

>whatever the Windows equivalent of /dev/null is

not sure that we have these equivalents in c#

Also, this change looks similar to how it's implemented in java: https://github.com/mongodb/mongo-java-driver/commit/9d0cac6972d0f25e5c986bd9db3492f5127ed23a#diff-44d9d1d766974cf831a080bfd1861ce3R54

>Do we need to worry about where standard out and standard error are pointing?

so, I would answer, no :)

Reply:
> /dev/null
Fwiw, for the "DOS" command prompt we have `NUL`, for Powershell we have `$null`
https://stackoverflow.com/questions/313111/is-there-a-dev-null-on-windows 
😸

Reply:
My concern is that standard output and standard error might be accumulating in a memory stream somewhere which would result in ever increasing memory use.

But I haven't been able to find a way to redirect output and *discard* it.

Reply:
Ah a good point!
A quick and dirty way may involve hooking into this event:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.outputdatareceived?redirectedfrom=MSDN&view=netframework-4.8
And discarding all the data.

There seem to be a lot of warnings surrounding using this stream regarding deadlocks, so it'd definitely behoove us to investigate this more quickly. A quick google search seems to indicate that this stream might become become too full at some point and block when output is generated. 

Seconding Robert's request for additional research here :)

Reply:
For now, I see two options:
1. Almost the same as already existed:

        using (Process process = new Process())
        {
            process.StartInfo.FileName = "path.exe";
            process.StartInfo.CreateNoWindow = true;
            process.StartInfo.UseShellExecute = false;
            process.StartInfo.RedirectStandardOutput = true;
            process.Start();

            process.StandardOutput.Dispose();
        }
This option is based on a comment in the dispose method for a process from github: https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L857
// Only call close on the streams if the user cannot have a reference on them.
// If they are referenced it is the user's responsibility to dispose of them.
I assume that it means that it's our responsibility to dispose of a streamReader for SyncMode (when we use `StandardOutput` directly).
I didn't find any other place(except in `Dispose` for async mode see below) in the github source where I could see a call of:

    _standardOutput ?.Dispose()

I don't think that developers accidentally forgot to do it.
Also, not sure that it makes any sense to continue provide any new output message after the time when the process is disposed. Who will be a receiver of these messages?
Also, there is a comment in the source: https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L804
    
    //Dispose managed and unmanaged resources

I don't know any reason to leave any inner streams alive after the time when we disposed unmanaged resourses..


2. The second option is almost the same but for async mode(when we need to call `BeginOutputReadLine` and able to use `OutputDataReceived` event). In this case, we don't need to call:
    
     process.StandardOutput.Dispose();

the streamReader will be disposed together with a process instance:
https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs#L868

Reply:
not sure what else i can do here

Reply:
Hm, how about this? 
`powershell -Command "mongocryptd 2>&1|out-null"`
`powershell -Command "mongocryptd 2>&1>$null"`

Reply:
Dima pointed out that Powershell might not be on the user's machine. Powershell is installed on all versions of windows starting with Windows 7

https://docs.microsoft.com/en-us/powershell/scripting/install/installing-windows-powershell?view=powershell-6
> Windows PowerShell comes installed by default in every Windows, starting with Windows 7 SP1 and Windows Server 2008 R2 SP1

but if we really want to support older versions of windows, this also works:
`cmd.exe /c "mongocryptd 2>&1>NUL"`

Reply:
@vincentkam basically you are proposing that instead of launching `mongocryptd.exe` directly we instead launch a shell (powershell or cmd) and have the shell launch `mongocryptd.exe`  for us using shell specific syntax to redirect output for us.

That shouldn't be necessary, and of course is platform specific.

Reply:
Added `--logpath nul --logappend` option for mongocryptd process.